src/engine/thread_pool.h

```diff
@@ -122,7 +123,7 @@ class ThreadPool {
   /*!
    * \brief Startup synchronization objects
    */
-  std::list<std::shared_ptr<SimpleEvent>> ready_events_;
+  std::vector<std::shared_ptr<SimpleEvent> > ready_events_;
```

Comment:
Are you sure about this? I think this is by design

Reply:
I checked the code several times, why use a list if we are just appending items, which are in addition pointers. I don't think it's a critical code path, correct me if I'm wrong, but in general a vector is faster,smaller and cache efficient.  Do you see any problems with this?

Reply:
Checked again, should be fine.

Reply:
Why did you change to vector?

Reply:
This is by design

Reply:
This is by design

Reply:
Inserting arbitrarily into a vector can cause CPU spikes. These are not accessed in a way that cache matters and random access is not required. Please revert.